kitsune/sumo/views.py

```diff
@@ -156,3 +157,35 @@ def serve_cors(*args, **kwargs):
     from django.views.static import serve
 
     return serve(*args, **kwargs)
+
+
+@require_GET
+def translate_url(request):
+    """Find the correct translation URL for a given document.
+
+    Required parameters:
+    - current_slug
+    - current_locale
+    - target_locale
+    """
+    current_slug = request.GET.get("current_slug")
+    current_locale = request.GET.get("current_locale")
+    target_locale = request.GET.get("target_locale")
+
+    if not all([current_slug, current_locale, target_locale]):
+        return JsonResponse({"error": "Missing required parameters", "found": False}, status=400)
+
+    try:
+        current_doc = Document.objects.get(slug=current_slug, locale=current_locale)
```

Comment:
I would add `.select_related("parent")` to this, just to avoid the possible extra query for the parent below.

Reply:
I just remembered another thing. You should used `Document.objects.visible(...` instead, because we shouldn't "leak" info about articles the user has no permission to see.